anchor-reference-server/src/main/java/org/stellar/anchor/reference/service/UniqueAddressService.java

```diff
@@ -0,0 +1,39 @@
+package org.stellar.anchor.reference.service;
+
+import java.util.Objects;
+import org.springframework.stereotype.Service;
+import org.stellar.anchor.api.callback.GetUniqueAddressResponse;
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.reference.config.AppSettings;
+import org.stellar.anchor.util.MemoHelper;
+
+@Service
+public class UniqueAddressService {
+  String distributionWallet;
+  String distributionWalletMemo;
+  String distributionWalletMemoType;
+
+  UniqueAddressService(AppSettings appSettings) throws SepException {
+    this.distributionWallet = appSettings.getDistributionWallet();
+    if (!Objects.toString(appSettings.getDistributionWalletMemo()).isEmpty()) {
```

Comment:
💡  Maybe it's worth testing if `distributionWallet` is a valid public key.

Reply:
Sure! Added!